transformations/multi_pivot_paraphrases_generation/transformation.py

```diff
@@ -0,0 +1,308 @@
+
+from transformers import MarianMTModel,MarianTokenizer
+
+from .easy_nmt import load_easynmt_model,get_easynmt_translation
+from .use_filter import load_use_model,get_use_embedding
+from .constants import HUGGINGFACE_MARIANMT_MODELS_TO_LOAD, EASYNMT_MODEL_NAME
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+# methods to print colored text in the terminal
+def pr_green(text):
+    """ Pring text in green color font"""
+    print("\033[92m{}\033[00m" .format(text))
+
+def pr_gray(text):
+    """ Pring text in gray color font"""
+    print("\033[7m{}\033[00m" .format(text))
+
+class MultiPivotParaphrasesGeneration(SentenceOperation):
+    """
+    This transformation generates a list of paraphrases for an English sentence by leveraging Pivot-Transaltion approach.
+    Pivot-Transaltion is an approach where a sentence in a source language is translated to a foreign language called the pivot language then
+    translated back to the source language to get a paraprhase candidate, e.g. translate an English sentence to French, then translate back to English.
+
+    The paraphrases generation is divided into two step:
+    - Step 1: paraphrases Candidate Over-generation by leveraging Pivot-Transaltion. At this step, we generate a Pool of possible parparhases.
+    - Step 2: apply a candidate selection over the Pool of paraphrases, since the pool can contain semantically unrelated or duplicate paraphrases.
+      We leverage Embedding Model such as Universal Sentence Encoder~(USE) to disqualify candidate paraphrases from the pool, by computing the Cosine Similarity socres of the
+      USE Embeddings between the reference sentence and the candidate paraphrase. Let R = USE_Embeding(reference_english_sentence) and P = USE_Embeding(candidate):
+        - if Cosine(R,P) < alpha => the candidate is semantically unrelated and then removed from the final list of paraphrases
+        - if Cosine(R,P) > beta => the candidate is a duplication and then removed from the final list of paraphrases
+        - By default Alpha=0.5 and Beta=0.95, we set the value as suggested by [Parikh et al.](https://arxiv.org/pdf/2004.03484.pdf) works
+    
+    Please refer to the test.json for all of the test cases catered.
+    
+    This transformation translates an English sentence to a list of predefined languages using Huggingface MariamMT and EasyNMT as Machine Transaltion models.
+    - The transformation support Two Pivot-Transaltion Level.
+        - If Pivot-level = 1 => Transalte to only one foreign language. e.g. English -> French -> English  ||  English -> Arabic -> English  ||  English -> japanese -> English
+        - If Pivot-level = 2 => Transalte to only Two foreign language. e.g. English -> French -> Arabic -> English  ||  English -> Russian -> Chinese -> English
+    """
+    
+    tasks = [
+        TaskType.QUESTION_GENERATION,
+        TaskType.TEXT_TO_TEXT_GENERATION
+    ]
+    languages = ["en"]
```

Comment:
Please add the relevant [keywords](https://github.com/GEM-benchmark/NL-Augmenter/blob/main/docs/keywords.md) here.

Reply:
Also in the end you can add a heavy=True parameter too.

Reply:
you are right. I would add the **lexical** and **syntactical** key word